impl/src/main/java/org/jboss/weld/bean/proxy/ProxyFactory.java

```diff
@@ -911,4 +918,17 @@ protected void getMethodHandlerField(ClassFile file, CodeAttribute b) {
     protected Class<?> getProxySuperclass() {
         return getBeanType().isInterface() ? Object.class : getBeanType();
     }
+
+    private static String getDefaultPackageReason(Class<?> clazz) {
+        if (clazz.getPackage() == null) {
+            return NULL;
+        }
+        if (clazz.getSigners() != null) {
+            return SIGNED;
+        }
+        if (clazz.getPackage().isSealed()) {
```

Comment:
Ok, did the test in a separate [branch](https://github.com/manovotn/core/blob/weld2402_test/tests-arquillian/src/test/java/org/jboss/weld/tests/proxy/sealed/ProxyCreationInSealedPackageTest.java) and it works without any code changes so I believe we should revert them in this PR. I will enrich this test to use some package private things and add a commit onto this PR.